torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -176,7 +179,8 @@ class TORCH_API ProcessGroupNCCL : public Backend {
     // In case of timeout, set exception on the WorkNCCL object.
     bool checkTimeout(
         c10::optional<std::chrono::milliseconds> timeout = c10::nullopt);
-
+    bool checkDesyncTimeout(
+        c10::optional<std::chrono::milliseconds> timeout = c10::nullopt);
```

Comment:
Is this used somewhere or do you plan to add two-tier timeout here?

Reply:
ohh good catch this is not intended to land